api/src/main/java/org/openmrs/module/initializer/api/InitializerServiceImpl.java

```diff
@@ -70,9 +71,8 @@ public String getChecksumsDirPath() {
 	
 	@Override
 	public List<Loader> getLoaders() {
-		List<Loader> loaders = Context.getRegisteredComponents(Loader.class);
-		Collections.sort(loaders);
-		return loaders;
+		return Context.getRegisteredComponents(Loader.class).stream().filter(l -> !l.isPreLoader()).sorted()
+		        .collect(Collectors.toList());
```

Comment:
See here how pre-loaders are **not** mixed up with loaders. This is why `Loader#isPreLoader()` was added.

Reply:
Ok..